webClient/src/app/services/FTAActivity.service.ts

```diff
@@ -0,0 +1,199 @@
+// /*
+//   This program and the accompanying materials are
+//   made available under the terms of the Eclipse Public License v2.0 which accompanies
+//   this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+//   SPDX-License-Identifier: EPL-2.0
+//   Copyright Contributors to the Zowe Project.
+// */
+import { Http, Headers, RequestOptions } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+import { Injectable, Inject } from '@angular/core';
+import { Angular2InjectionTokens } from 'pluginlib/inject-resources';
+import { of } from 'rxjs';
+import * as globals from '../../environments/environment';
+
+@Injectable()
+export class FTAActivityService {
+  private scope: string = 'user';
+  private resourcePath: string = 'ui/ftaactivity';
+  private basePlugin: ZLUX.Plugin;
+
+  private resourceName: string;
+  private uri: string;
+  public fatDownloadActivity: FTAActivity[];
+  public fatDownloadActivityCancelled: FTAActivity[];
+  public fatDownloadActivityCompleted: FTAActivity[];
+  public fatUploadActivity: FTAActivity[];
+  public fatTransferActivity: FTAActivity[];
+  private initFTAActivity:boolean; 
+  private type:string;
+  private limitofActivityHistory:number;
+  public config = globals.prod_config;
+
+  constructor(@Inject(Angular2InjectionTokens.PLUGIN_DEFINITION) private pluginDefinition: ZLUX.ContainerPluginDefinition
+  , @Inject(Angular2InjectionTokens.LOGGER) private log: ZLUX.ComponentLogger
+  , private http: Http) {}
+
+  onInit() {
+    this.limitofActivityHistory = 10;
+    this.basePlugin = this.pluginDefinition.getBasePlugin();
+    this.resourceName = `activity.json`;
+    this.uri = ZoweZLUX.uriBroker.pluginConfigForScopeUri(this.basePlugin, this.scope, this.resourcePath, this.resourceName);
+    this.fatUploadActivity = [];
+    this.fatDownloadActivity = [];
+    this.fatDownloadActivityCompleted = [];
+    this.fatDownloadActivityCancelled = [];
+    this.fatTransferActivity = [];
```

Comment:
I believe, the only things that really need to go in onInit are lines 39, and maybe line 41

Reply:
The rest can go in constructor right?

Reply:
exactly